File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function (nums) {
  const all = [];
  const map = new Map();
  for (let i in nums) {
    map.set(nums[i], (map.get(nums[i]) ?? 0) + 1);
  }
  for (let i = 1; i <= nums.length; i++) {
    // console.log(map.get(i))
    if (!map.get(i)) {
      all.push(i);
    }
  }
  return all;
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {

    public List<Integer> findDisappearedNumbers(int[] nums) {
        ArrayList<Integer> arr = new ArrayList<>();
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }

        for (int i = 1; i <= nums.length; i++) {
            // System.out.print(map.get(i)+" ");
            if (map.get(i) == null) {
                arr.add(i);
            }
        }
        return arr;
    }
}
